packages/core/src/lib/utils/env.ts

```diff
@@ -51,13 +51,22 @@ export function createActionURL(
   envObject: any,
   basePath?: string
 ): URL {
-  let url = envObject.AUTH_URL ?? envObject.NEXTAUTH_URL
-  if (!url) {
-    const host = headers.get("x-forwarded-host") ?? headers.get("host")
-    if (!host) throw new TypeError("Missing host")
-    const proto = headers.get("x-forwarded-proto") ?? protocol
-    url = `${proto === "http" ? "http" : "https"}://${host}${basePath}`
+  let envUrl = envObject.AUTH_URL ?? envObject.NEXTAUTH_URL
+
+  let detectedHost = headers.get("x-forwarded-host") ?? headers.get("host")
+  let detectedProtocol = headers.get("x-forwarded-proto") ?? protocol ?? "https"
```

Comment:
these are only used in the else block - I suggest we can move them there and turn into `const`

Reply:
Done